c7n/resources/glue.py

```diff
@@ -272,3 +272,122 @@ def process(self, resources):
+        enum_spec = ('get_classifiers', 'Classifiers', None)
+        id = name = 'Name'
+        date = 'CreationTime'
+        arn_type = 'classifier'
+
+    permissions = ('glue:GetClassifiers',)
+
+
+@GlueClassifier.action_registry.register('delete')
+class DeleteClassifier(BaseAction):
+
+    schema = type_schema('delete')
+    permissions = ('glue:DeleteClassifier',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('glue')
+        for r in resources:
+            classifier = next(iter(r.values()))
+            try:
+                client.delete_classifier(Name=classifier['Name'])
+            except client.exceptions.EntityNotFoundException:
+                continue
+
+
+@resources.register('glue-security-configuration')
+class GlueSecurityConfiguration(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'glue'
+        enum_spec = ('get_security_configurations', 'SecurityConfigurations', None)
+        id = name = 'Name'
+        arn_type = 'securityConfiguration'
+        date = 'CreatedTimeStamp'
+
+    permissions = ('glue:GetSecurityConfigurations',)
+
+
+@GlueSecurityConfiguration.action_registry.register('delete')
+class DeleteSecurityConfiguration(BaseAction):
+
+    schema = type_schema('delete')
+    permissions = ('glue:DeleteSecurityConfiguration',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('glue')
+        for r in resources:
+            try:
+                client.delete_security_configuration(Name=r['Name'])
+            except client.exceptions.EntityNotFoundException:
+                continue
+
+
+@resources.register('glue-trigger')
+class GlueTrigger(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'glue'
+        enum_spec = ('get_triggers', 'Triggers', None)
+        id = name = 'Name'
+        arn_type = 'trigger'
+        universal_taggable = object()
+
+    permissions = ('glue:GetTriggers',)
+    augment = universal_augment
+
+
+@GlueTrigger.action_registry.register('delete')
+class DeleteTrigger(BaseAction):
+
+    schema = type_schema('delete')
+    permissions = ('glue:DeleteTrigger',)
+
+    def process(self, resources):
+        client = local_session(self.manager.session_factory).client('glue')
+        for r in resources:
+            try:
+                client.delete_trigger(Name=r['Name'])
+            except client.exceptions.EntityNotFoundException:
+                continue
+
+
+@resources.register('glue-workflow')
+class GlueWorkflow(QueryResourceManager):
+
+    class resource_type(TypeInfo):
+        service = 'glue'
+        enum_spec = ('list_workflows', 'Workflows', None)
+        detail_spec = ('get_workflow', 'Name', None, 'Workflow')
+        id = name = 'Name'
+        arn_type = 'workflow'
+        universal_taggable = object()
+
+    permissions = ('glue:ListWorkflows',)
+
+    def augment(self, resources):
```

Comment:
augment = universal_augment at class and avoid the method unless its doing/delegating something useful

Reply:
I need this method to avoid running into errors when iterating through the resources here, stemming from a problem with augment: https://github.com/cloud-custodian/cloud-custodian/blob/f5c5272b5955fde668cd4a7921b254b43016dff3/c7n/query.py#L543-L551. Also, the firehose resource which makes use of `detail_spec` uses the same construct: https://github.com/cloud-custodian/cloud-custodian/blob/f5c5272b5955fde668cd4a7921b254b43016dff3/c7n/resources/kinesis.py#L108-L110